File: Data Structures & Algorithms/invert-a-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public TreeNode invertTree(TreeNode root) {
        if (root == null)   {
            return null;
        }
        return invertRecursive(root);
    }

    private TreeNode invertRecursive(TreeNode curr) {
        if (curr != null)   {
            TreeNode val1 = invertRecursive(curr.left);
            TreeNode val2 = invertRecursive(curr.right);
            curr.left = val2;
            curr.right = val1;
        }
        return curr;
    }
}

File: Data Structures & Algorithms/balanced-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        self.is_balanced = True
        self.dfs(root)

        return self.is_balanced

    def dfs(self, curr):
        if not curr:
            return 0
        
        left = self.dfs(curr.left)
        right = self.dfs(curr.right)
        diff = abs(right - left)

        if diff > 1:
            self.is_balanced = False

        return 1 + max(left, right)
        